virl2_client/models/node.py

```diff
@@ -404,15 +405,48 @@ def boot_disk_size(self, value: int) -> None:
 
     @property
     def configuration(self) -> str | None:
-        """Return the initial configuration of this node."""
-        self.lab._sync_topology(exclude_configurations=False)
-        return self._configuration
+        """Return the contents of the main configuration file."""
+        self.lab.sync_topology_if_outdated(exclude_configurations=False)
+        return self._configuration[0].get("content") if self._configuration else None
 
     @configuration.setter
-    def configuration(self, value) -> None:
-        """Set the initial configuration of this node."""
+    def configuration(self, value: str | list | dict) -> None:
+        """
+        Set the content of:
+         - the main configuration file if passed a string,
+         - one configuration file if passed a dictionary in the format of
+        `{"name": "filename.txt", "content": "<file content>"}`,
+         - or multiple configuration files if passed a list of above dictionaries.
+        Can also use "Main" in place of the filename of the main configuration file.
+
+        :param value: The configuration data in one of three formats.
+        """
         self._set_node_property("configuration", value)
-        self._configuration = value
+        self._configuration = (
+            self._configuration if self._configuration is not None else []
+        )
+        if isinstance(value, str):
+            if not self._configuration:
+                self._configuration.append({"name": "Main", "content": value})
+            else:
+                self._configuration[0]["content"] = value
+            return
+        new_configs = value if isinstance(value, list) else [value]
+        new_configs_by_name = {
+            new_config["name"]: new_config for new_config in new_configs
+        }
+        for i, config in enumerate(self._configuration):
+            if config["name"] in new_configs_by_name:
+                self._configuration[i] = new_configs_by_name[config["name"]]
+
+    @property
+    def configuration_files(self) -> list[dict[str, str]] | None:
+        """
+        Return all configuration files, in a list in the following format:
+        `[{"name": "filename.txt", "content": "<file content>"}]`
+        """
+        self.lab.sync_topology_if_outdated(exclude_configurations=False)
+        return deepcopy(self._configuration)
```

Comment:
I'm wondering whether we should have two getters but just one setter?  What if we added a new parameter, e.g., all by default set to False to include all configs in the configuration getter?

Reply:
Can't add parameters to a getter; we'd have to change it to a function, which would not be backwards compatible, so for compatibility we'd have to rename the function and deprecate the getter... Yeah, no. Two getters are not really a problem. It's quite clear which returns what, and the setter has pretty clear typing hints to suggest that it accepts multiple types of input, so there should not be any confusion.